components/brave_wallet_ui/components/buy-send-swap/swap-input-component/index.tsx

```diff
@@ -139,22 +145,28 @@ function SwapInputComponent (props: Props) {
       {componentType !== 'selector' &&
         <>
           <Row>
-            <FromBalanceText isExchange={componentType === 'exchange'}>{getTitle()}</FromBalanceText>
-            {componentType === 'exchange' ? (
+            <FromBalanceText componentType={componentType}>{getTitle()}</FromBalanceText>
+            {componentType === 'exchange' &&
               <MarketLimitButton onClick={onToggleOrderType}>{orderType === 'market' ? locale.swapLimit : locale.swapMarket}</MarketLimitButton>
-            ) : (
+            }
+            {componentType !== 'exchange' && componentType !== 'toAddress' &&
               <FromBalanceText>{locale.balance}: {selectedAssetBalance}</FromBalanceText>
-            )
+            }
+            {componentType === 'toAddress' &&
+              <PasteButton onClick={onPaste}>
+                <PasteIcon />
+              </PasteButton>
             }
           </Row>
-          <Row isExchange={componentType === 'exchange'}>
+          <Row componentType={componentType}>
             <Input
-              isExchange={componentType === 'exchange'}
-              type='number'
-              placeholder='0.0'
-              value={selectedAssetInputAmount}
+              componentType={componentType}
+              type={componentType === 'toAddress' ? 'text' : 'number'}
```

Comment:
Not for this PR, but I can see this file quickly becoming messy due to different `componentType` values. It's a downside of sharing presentational React code. I'm not sure if there's an elegant solution for this, but maybe we could have generic components that are not aware of `componentType`.